### PR TITLE
Support manifest v3 in Chrome

### PR DESCRIPTION
**Pull request by eetann**
**English version of the text follows below the Japanese** 🙏 

---
# 概要
Chrome拡張機能は、2023年1月までにmanifest v2からv3に移行する必要があります。このプルリクエストはその移行の対応をしました。

manifest v3への移行について、いくつかリンクを載せます。もしご存知でしたら申し訳ございません。

- [Chromeでのmanifest v2のサポート期限についての記事](https://developer.chrome.com/docs/extensions/mv3/mv2-sunset/)
- [「Google Chrome 88」がリリース ～“Manifest V3”で拡張機能はより安全・高速に【1月22日追記】 - 窓の杜](https://forest.watch.impress.co.jp/docs/news/1301206.html)
- [Migrating to Manifest V3 - Chrome Developers](https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/)

# 変更の詳細
公式ドキュメントに掲載されている移行手順 [Migrating to Manifest V3 - Chrome Developers](https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/) に沿って変更しました。変更の概要は以下のとおりです。

- `manifest_version` を2から3へ
    - 参考：[Manifest version](https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/#manifest-version)
- `Service worker` として動くようにmanifest・scriptを変更
    - Service workerではlocalStorageが使用できないようなので、本拡張機能で書かれていたchrome.storageのラッパー`LocalStorageRepository` を使って書き換えました
    - 参考：[Service worker1](https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/#man-sw)、[Service workers2](https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/#background-service-workers)
- Action APIの統合
    - `browser_action` と`page_action` が統合され、`action` になりました。それに伴ってmanifest・APIともに変更しています
    - 参考：[Action API unification](https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/#action-api-unification)
- Web-accessible resourcesの変更
    - 参考：[Web-accessible resources](https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/#web-accessible-resources)
- 廃止されたAPIの置換
    - `chrome.extension.getURL`→ `chrome.runtime.getURL`
    - 参考：[Sunset for deprecated APIs](https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/#sunset-deprecated-apis)

# その他
- このPRでは機能変更は行っていません。Chromeのmanifest v3対応のみです
- `npm run test` → 31 passing
- 私の手元にあるMacのChrome・Firefoxでの動作確認は行いましたが、できればtilfinさんの手元の環境でも動作確認をお願いしたいです

---

# Overview
Chrome extensions must migrate from manifest v2 to v3 by January 2023. This pull request addresses that migration.

I will post some links regarding the migration to manifest v3. If you know of any, I apologize.

- [article about the expiration of manifest v2 support in Chrome](https://developer.chrome.com/docs/extensions/mv3/mv2-sunset/)
- [「Google Chrome 88」がリリース ～“Manifest V3”で拡張機能はより安全・高速に【1月22日追記】 - 窓の杜](https://forest.watch.impress.co.jp/docs/news/1301206.html)
- [Migrating to Manifest V3 - Chrome Developers](https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/)

# Change Details
Migration instructions in the official documentation [Migrating to Manifest V3 - Chrome Developers](https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/). Here is a summary of the changes

- `manifest_version` from 2 to 3
  - Reference: [Manifest version](https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/#manifest-version)
- Changed manifest & script to work as `Service worker`
    - Since localStorage does not seem to be available for Service workers, I rewrote it using the `LocalStorageRepository` wrapper for chrome.storage that was written in this extension.
    - Reference: [Service worker1,](https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/#man-sw) [Service workers2](https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/#background-service-workers)
- Action API unification
    - The `browser_action` and `page_action` have been merged into `action`. Therefore, both manifest and API have been changed.
    - Reference: [Action API unification](https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/#action-api-unification)
- Web-accessible resources has been changed
  - Reference: [Web-accessible resources](https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/#web-accessible-resources)
- Replacement of deprecated APIs
    - `chrome.extension.getURL` -> `chrome.runtime.getURL`.
    - Reference: [Sunset for deprecated APIs](https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/#sunset-deprecated-apis)

# etc.
- This PR does not change functionality, only Chrome manifest v3 support.
- `npm run test` → 31 passing
- I have tested it with Chrome and Firefox on my Mac, but I'd appreciate it if you could check it with your own environment if possible

### CODE REVIEW

File: src/popup.js [hunk 1]
```diff
@@ -169,10 +174,13 @@ function setupRoleFilter() {
...
-    let swcnt = localStorage.getItem('switchCount') || 0;
-    localStorage.setItem('switchCount', ++swcnt);
-    window.close()
+    localStorageRepo.get(['switchCount'])
+      .then(data => {
+        let swcnt = data.switchCount || 0;
+        localStorageRepo.set({ switchCount: ++swcnt }).then(() => {})
+      })
   });
+  window.close()
```
Comment (tilfin) [inline src/popup.js:183]: Why did you move this line out of the promise chain?

Comment (tilfin) [reply src/popup.js:183]: It is wrong solution so please revert last commit. This problem isn't related to manifest v3.

Comment (tilfin) [reply src/popup.js:183]: It is fixed by #267 .

Comment (eetann) [reply src/popup.js]: The problem was caused by a different callback, sorry. I deleted the last commit.

**Review by tilfin — APPROVED** [review]
LGTM! Thank you.
